py/desispec/scripts/daily_processing.py

```diff
@@ -0,0 +1,239 @@
+
+                prow = erow_to_prow(erow)
+
+                if camword is not None:
+                    prow['CAMWORD'] = camword
+                prow['INTID'] = internal_id
+                internal_id += 1
+                prow = define_and_assign_dependency(prow, arcjob, flatjob)
+                print(f"Processing: {prow}")
+                prow = create_and_submit(prow, dry_run=dry_run, queue=queue)
+                ptable.add_row(prow)
+
+                if curtype == 'flat' and flatjob is None and int(erow['SEQTOT']) < 5:
+                    flats.append(prow)
+                elif curtype == 'arc' and arcjob is None:
+                    arcs.append(prow)
+                elif curtype == 'science' and last_not_dither:
+                    sciences.append(prow)
+
+                lasttile = curtile
+                lasttype = curtype
+                last_not_dither = (prow['OBSDESC'] != 'dither')
+
+            time.sleep(10)
+            write_tables( [etable, ptable, unproc_table],
+                          fullpathnames=[exp_table_pathname,proc_table_pathname,unproc_table_pathname] )
+
+        time.sleep(300*speed_modifier)
+
+        if len(ptable) > 0:
+            ptable, nsubmits = update_and_recurvsively_submit(ptable,start_time=nersc_start,end_time=nersc_end,
+                                                              ptab_name=proc_table_pathname, dry_run=dry_run)
+
+            ## Exposure table doesn't change in the interim, so no need to re-write it to disk
+            write_table(ptable, tablename=proc_table_pathname)
+            time.sleep(300*speed_modifier)
+        
+    ## No more data coming in, so do bottleneck steps if any apply
+    ptable, arcjob, flatjob, internal_id = checkfor_and_submit_joint_job(ptable, arcs, flats, sciences, \
+                                                                         flatjob, arcjob, lasttype, last_not_dither,\
+                                                                         internal_id, dry_run=dry_run, queue=queue)
+
+
+    ## All jobs now submitted, update information from job queue and save
+    ptable = update_from_queue(ptable,start_time=nersc_start,end_time=nersc_end, dry_run=dry_run)
+    write_table(ptable, tablename=proc_table_pathname)
+
+    ## Now we resubmit failed jobs and their dependencies until all jobs have un-submittable end state
+    ## e.g. they either succeeded or failed with a code-related issue
+    ii,nsubmits = 0, 0
+    while ii < 4 and continue_looping(ptable['STATUS']):
+        print(f"Starting iteration {ii}")
+        ptable, nsubmits = update_and_recurvsively_submit(ptable, submits=nsubmits, start_time=nersc_start,end_time=nersc_end,
+                                                          ptab_name=proc_table_pathname, dry_run=dry_run)
+        write_table(ptable, tablename=proc_table_pathname)
+        if continue_looping(ptable['STATUS']):
+            time.sleep(1800*speed_modifier)
```

Comment:
This is a really long sleep; could it just be the standard sleep time?

Reply:
This is monitoring and resubmitting queue jobs. Arc jobs take 30+ minutes. Maybe two or three flats/sciences could run in that time, so I don't think it's unrealistic.

Reply:
Can now be user specified, and I think there was a mixup in the use of this sleep that is hopefully now more clear.